src/blog/2024/07/building-on-flowfuse-devices.md

```diff
@@ -0,0 +1,103 @@
+- **Parse:** Process the data into a computational format
+- **Compute/Monitor:** Analysis & monitoring of the data from all devices
+- **Action:** Taking action based on the data from the devices
+
+Then, we have to consider that this functionality needs to be deployed out to thousands of devices. So, our final requirement is:
+
+- **Maintain:** If we update functionality on one device, we may need to be able to easily roll those updates out to other devices. We need to sure it's easy to update and deploy to our devices, and if anything goes wrong, we should be able to easily find out what the problem is, and get easy access to fix it.
+
+## Architecture
+
+Let's consider an example architecture set in an automotive plant:
+
+![Diagram showing the architecture of a "Device Monitoring" use case in an automotive plant.](./images/device-monitoring-architecture.png){data-zoomable}
+_Diagram showing an example architecture of a "Device Monitoring" use case in an automotive plant._
+
+- **Body Shop:** Here we have several piece of machinery, each with the "FlowFuse Device Agent" installed. This allows us to manage the Node-RED deployments on these machines remotely from FlowFuse, and easily extract data from the machines for analysis.
+- **Plant:** We have multiple servers running at the Planet-level, generally one for each "Shop", each with their own Device Agent installed, again for easy remote management and deployment of Node-RED, e.g. Dashboards that provide a single HMI for monitoring hardware across a full shop. Here, we also have our instance of FlowFuse. This is the central point for managing all of our Node-RED deployments across the factory floor.
+- **Company IT Dept:** The general IT department of the company would provide multiple servers and services, accessible to a range of departments and the plant. Node-RED could act as a bridge between the Plant and the Company IT Dept, allowing us to easily extract data from the Plant and send it to the Company IT Dept, and vice-versa.
+- **Cloud:** Here we demonstrate how a company may have external Cloud-based services they're dependent upon. For example, image analytics services. Node-RED can be used to parse machine imagery, and integrate straight with these external services from the FlowFuse server.
+
+This diagram also demonstrates a sample of the rich ecosystem of communication protocols that Node-RED can support:
+
+- **TCP:** Show here to interface between...
+- **EtherNet/IP:** Insert example of what this would be used for in such a use case...
+- **MQTT:**  Insert example of what this would be used for in such a use case...
+- **OPC-UA:**  Insert example of what this would be used for in such a use case...
+- **Modbus:**  Insert example of what this would be used for in such a use case...
+- **MC Protocol:**  Insert example of what this would be used for in such a use case...
+- **Siemens S7:**  Insert example of what this would be used for in such a use case...
```

Comment:
```suggestion
- **TCP:** Read and process data from from bespoke and custom devices
- **EtherNet/IP:** Collect data from Allen Bradley and other EtherNet/IP devices
- **MQTT:**  Modern devices and factory generated IIoT data
- **OPC-UA:**  Communicate with newer PLCs and OPC Servers 
- **Modbus:**  Data collection from existing Modbus enabled devices like Temperature Probes, Invertors, Encoders
- **MC Protocol:**  Gather data from Mitsubishi PLCs
- **Siemens S7:**   Gather data from Siemens PLCs using the S7 Protocol
```

Not sure what angle to come at this with Joe - wrote something that can be iterated.

Reply:
Was going to try and tailor per the diagram we put together, and an example of how it'd fit into an automative plant